File: Mona_py/Commands/JS/autocrop.js
const Discord = require('discord.js');
const main = require('../index.js');
const Jimp = require('jimp');

module.exports = {
   name: 'autocrop',
   description: 'Automatically crops transparent pixels from a PNG',
   async execute(message, args) {
      console.command(`${"!mona autocrop".magenta} in ${(message.guild.name).toString().magenta}`)
      
      let imageURL = message.attachments.map(attachment => attachment.url)[0];
      if (imageURL == undefined) return message.channel.send(`Please attach an image alongside the command!`); 
      let image = await Jimp.read(imageURL);
      image.autocrop(0.1);
      image.write(`./temp/autocropped_${message.id}.png`); 
      
      const attachment = new Discord.MessageAttachment(`./temp/autocropped_${message.id}.png`);
      try {
         await message.channel.send(attachment); 
      } catch (error) {
         if (error) message.channel.send(`${message.author} I don't have permissions to attach files!`);
      }
   },
};

File: Mona_py/Commands/autocrop.py
import nextcord

from Mona_py.utils import log
from PIL import Image
from io import BytesIO


async def autocrop(interaction: nextcord.Interaction, attachment: nextcord.Attachment):
    log("command", f"Slash command /autocrop used in {interaction.guild.name}")

    if not attachment:
        await interaction.response.send_message("Please attach an image alongside the command!")
        return

    image_data = await attachment.read()

    try:
        with Image.open(BytesIO(image_data)) as img:
            img = img.convert("RGBA")           # Ensure image is in RGBA format
            img = img.crop(img.getbbox())       # Crop the image

            output = BytesIO()
            img.save(output, format='PNG')
            output.seek(0)

            file = nextcord.File(output, f'autocropped_{interaction.id}.png')
            await interaction.response.send_message(file=file)

    except Exception as e:
        await interaction.response.send_message(f"Error processing the image: {str(e)}")